torch/_inductor/kernel/conv.py

```diff
@@ -308,8 +308,20 @@ def convolution(
     dilation = pad_listlike(dilation, ndim)
     output_padding = pad_listlike(output_padding, ndim)
 
+    def channels_last_conv():
+        if V.graph.layout_opt and ndim == 2:
+            return True
+
+        layout = conv_layout(x, weight, None, **kwargs)
+        req_stride_order = ir.get_stride_order(
+            V.graph.sizevars.size_hints(layout.stride)
+        )
+        return req_stride_order == ir.NHWC_STRIDE_ORDER
```

Comment:
just to make sure I understand -- this is checking that the layout of the output is NHWC, and if it is, then `x` and `w` are both NHWC?

Reply:
When we lower conv1x1 as mm, we are making the output NHWC. This is checking that we would have made it NHWC anyway, by looking at the output layout. 

When you call into `aten.convolution`, if either `x` or `w` is channels last, then the other will be made to be channels last as well with the op, then the kernel is invoked.

> x and w are both NHWC 

So, this is true with respect to the convolution kernel itself, but not necessarily the inputs to `aten::convolution`. Just one input needs to be NHWC for the output to be NHWC.

Within the conv1x1 as mm function, we force the input to be channels last strides. So we want to avoid :

- an extra copy to the input making it channels last where the copy wouldn't have happened otherwise
- accidentally changing the output striding, which might have downstream effects